src/WebJobs.Script/Config/TransmissionStatusTelemetryModule.cs

```diff
@@ -0,0 +1,131 @@
+    internal class TransmissionStatusTelemetryModule : ITelemetryModule, IDisposable
+    {
+        private bool isInitialized = false;
+        private DiagnosticListener source = new DiagnosticListener(ScriptConstants.HostDiagnosticSourcePrefix + "ApplicationInsights");
+        private JsonSerializerSettings options = new JsonSerializerSettings()
+        {
+            Error = (sender, error) => error.ErrorContext.Handled = true,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+
+        /// <summary>
+        /// Initializes the telemetry module.
+        /// </summary>
+        /// <param name="configuration">Telemetry configuration to use for initialization.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            // Prevent the telemetry module from being initialized multiple times.
+            if (isInitialized)
+            {
+                return;
+            }
+            if (configuration.TelemetryChannel is ServerTelemetryChannel)
+            {
+                (configuration.TelemetryChannel as ServerTelemetryChannel).TransmissionStatusEvent += Handler;
+                isInitialized = true;
+            }
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        public void Dispose()
+        {
+            source.Dispose();
+        }
+
+        internal void Handler(object sender, TransmissionStatusEventArgs args)
+        {
+            if (args == null)
+            {
+                source.Write("TransmissionStatus", "Unable to parse transmission status response, args is null");
+                return;
+            }
+
+            // Always log if the response is non-success or if the feature flag is enabled
+            if (args.Response?.StatusCode != 200 || source.IsEnabled(ScriptConstants.HostDiagnosticSourceDebugEventNamePrefix))
+            {
+                source.Write("TransmissionStatus", FormattedLog(sender, args));
+            }
+        }
+
+        internal string FormattedLog(object sender, TransmissionStatusEventArgs args)
+        {
+            var transmission = sender as Transmission;
+            var items = transmission?.TelemetryItems.GroupBy(n => n.GetEnvelopeName())
+                        .Select(n => new
+                        {
+                            type = n.Key,
+                            count = n.Count()
+                        });
+
+            BackendResponse backendResponse = null;
+            if (!string.IsNullOrWhiteSpace(args?.Response?.Content))
+            {
+                backendResponse = JsonConvert.DeserializeObject<BackendResponse>(args.Response.Content, options);
+            }
+
+            string topErrorMessage = null;
+            int? topStatusCode = null;
+            if (backendResponse?.Errors != null && backendResponse.Errors.Length > 0)
+            {
+                topErrorMessage = backendResponse.Errors[0].Message;
```

Comment:
We only log the first error message?. Would logging all the error messages be useful? May be build a comma separated string?

Reply:
The way we use AI SDK, the error message is going to be same for all the telemetry items. I am trying to keep the log message as small it can be.